src/bitcoin-miner.cpp

```diff
@@ -255,7 +255,7 @@ static UniValue CpuMineBlock(unsigned int searchDuration, const UniValue &params
 
     uint32_t startNonce = header.nNonce = std::rand();
 
-    printf("Mining: id: %lx parent: %s bits: %x difficulty: %3.2f time: %d\n", (uint64_t)params["id"].get_int64(),
+    printf("Mining: id: %x parent: %s bits: %x difficulty: %3.2f time: %d\n", (unsigned int)params["id"].get_int64(),
```

Comment:
why cast to unsigned int when you're returning an int64?  why not just print out the int64 using "ld"

Reply:
so it works on both 32 and 64 bit computers